tasks/main.yml

```diff
@@ -14,11 +19,41 @@
   when: users_create_per_user_group
   tags: ['users','configuration']
 
+- name: Creating groups
+  group:
+    name: "{{ item }}"
+  when: item.uid is defined
+  with_items: "{{ groups_to_create }}"
+  tags: ['users','groups','configuration']
+
+- name: Per-user group creation
+  group:
+    name: "{{ item.username }}"
+    gid: "{{ item.uid }}"
+  with_items: "{{ users }}"
+  when: users_create_per_user_group and item.uid is defined
+  tags: ['users','configuration']
+
 - name: User creation
   user:
     name: "{{ item.username }}"
+    group: "{{ users_create_per_user_group | ternary(item.username,users_group) }}"
+    groups: "{{ item.groups | join(',') }}"
+    shell: "{{ (item.shell is defined) | ternary(item.shell, users_default_shell) }}"
+    password: "{{ item.password | default('!') }}"
+    comment: "{{ item.name }}"
+    uid: "{{ item.uid }}"
     createhome: "{{ users_create_homedirs | bool }}"
-  with_items: "{{ users }}"
+  when: item.stdout == ""
```

Comment:
This stdout match is pretty strange. Can you have the command that registers this above, simply exit with a 0 or 1 and then use the `success` filter.

I'd rather this read something like:

```yaml
when: item | success
```